third_party/move/move-vm/runtime/src/storage/module_storage.rs

```diff
@@ -472,10 +460,24 @@ impl<'a> FunctionValueExtension for FunctionValueExtensionAdapter<'a> {
                         ty_builder.create_ty_with_subst(ty, &fun.ty_args)
                     }
                 };
+
+                let loader = NativeLoader::new(self.module_storage);
+                let layout_converter = LayoutConverter::new(&loader);
+
                 let captured_layouts = mask
                     .extract(fun.param_tys(), true)
                     .into_iter()
-                    .map(|t| ty_converter.type_to_type_layout(&instantiate(t)?))
+                    .map(|t| {
+                        // TODO(georgemitenkov):
+                        //   Double check this is correct and aggregators work with closures.
+                        layout_converter.type_to_type_layout(
+                            // We do not expect gas to be metered here. The only thing that is
```

Comment:
If this is always the case, we can create the corresponding checking traversal here instead of wrapping the caller? But the way it is now is more general, except for coupling with UnmeteredGasMeter here. Alternatively, we could maybe pass gas meter as well (or if it could be part of module traversal context as well)